specs/src/recovery.md

```diff
@@ -0,0 +1,525 @@
+
+Alone, distributing parity data during recovery only makes attacks more
+difficult. Both the pull based and broadcast tree attacks can still occur, they
+just require more malicious nodes.
+
+While additional mechanisms are still needed, adding parity data to the recovery
+process is a powerful tool to get to our goal:
+
+>Its important that new broadcasting mechanisms do not introduce a lower barrier
+>to halting the network than standard proof of stake already does.
+
+#### Overlay Networks
+
+An overlay network for validators means that:
+
+- **Validators can verifiably connect to and find other validators in the
+  network**
+
+We do not specify an overlay network further here. The benefit of an overlay
+network is simply that validators can avoiding sybil nodes to the best of their
+knowledge.
+
+If we can make the attacks require >= 1/3 of the voting power, arguably we have
+achieved our original goal.
+
+>Its important that new broadcasting mechanisms do not introduce a lower barrier
+>to halting the network than standard proof of stake already does.
+
+#### Preparation
+
+If we have the ability to know what portion of the voting power already has a
+given transaction, and the network is able to reuse the transactions they have
+already downloaded in the block, then we can use that to our advantage when
+combatting sybil attacks. 
+
+The more voting power that has signaled that they have a transaction, then the
+more difficult it is to stop the distribution of that transaction.
+
+### Synthesis
+
+Alone, none of the mechanisms listed above are sufficient for preventing sybil
+attacks. However, when we combine them we can meet our original goal:
+
+>Its important that new broadcasting mechanisms do not introduce a lower barrier
+>to halting the network than standard proof of stake already does.
+
+- If parity data is also gossiped, then the acceptable number of failures is
+  raised relative to the erasure encoding factor.
+- If an overlay network is added, then validators are able to avoid sybil
+  attacks from non-validators
+- If some portion of the proposed block is already distributed to a majority of
+  validators, then it has already been sufficiently distributed to bypass sybil.
+
+## Messages
+
+### Commitment
+
+```proto
+// TxMetaData contains all information needed to commit to a transaction. Start
+// and end are used to insert the transaction into the protobuf encoded version
+// of the block.
+message TxMetaData {
+  bytes  hash  = 1;
+  uint32 start = 2;
+  uint32 end   = 3;
+}
+
+// CompactBlock commits to the transaction included in a proposal. It enalbes
+// clients to reuse already downloaded blobs instead of gossiping them all again
+// during the block propagation.
+message CompactBlock {
+  bytes                     bp_hash      = 1;
+  repeated TxMetaData       blobs        = 2;
+  bytes                     signature    = 3;
+  tendermint.types.Proposal proposal     = 4;
```

Comment:
Why does the compact block also have the proposal? In the past the proposal and the block have been sent separately

Reply:
This is likely to contain height, round, and pol_round but also not sure.

Reply:
Ok but does that mean we only send the compact block (not a separate proposal). And if it's just for those fields, could those fields be added directly instead of the entire proposal struct?